core/ipc/sink/camel/client/src/main/java/org/opennms/core/ipc/sink/camel/client/CamelRemoteMessageDispatcherFactory.java

```diff
@@ -78,29 +79,31 @@
                 CamelSinkConstants.JMS_QUEUE_PREFIX, module.getId());
         Map<String, Object> headers = new HashMap<>();
         headers.put(CamelSinkConstants.JMS_QUEUE_NAME_HEADER, queueNameFactory.getName());
-        return headers;
+        return Collections.unmodifiableMap(headers);
```

Comment:
Since in every dispatch we are adding tracing headers, we can't have `UnmodifiableMap` here.   I had to change this when I was doing tracing related changes.

Reply:
Looks like you are creating new HashMap below

Reply:
Yeah, there have been some conditional adds to the list which result in leaving the provious value in the list. Therefore I wanted to ensure that the list is immutable